cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go

```diff
@@ -37,6 +37,53 @@ const (
 	NodesKubeadmConfigClusterRoleName = "kubeadm:nodes-kubeadm-config"
 )
 
+// UploadRemoveConfiguration uploads a new configuration with the client removed
+func UploadRemoveConfiguration(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
```

Comment:
What about calling this func `ResetClusterStatusForNode` instead of `UploadRemoveConfiguration`?
if possible, invert the order of parameters (first `client`, then `cfg`); additionally I the next comments apply, don't think we need all the cfg, but only the node name.

Reply:
+1 on that.